samtranslator/utils/traverse.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Callable, Dict, List
+
+
+class Traverse:
```

Comment:
I think ideally we would have an abstract `Resolver` class that has the generic traversing logics. Then we could have `IntrinsicsResolver` that inherits `Resolver` class. We would also have a `DependsOnResolver` class that inherids `Resolver` class.